src/systemtest/test_level2.py

```diff
@@ -322,89 +305,118 @@ def test_scanid_mismatch(self, odinapi_service, cleanup):
-            host=odinapi_service,
+        url = "{host}/rest_api/{version}/level2/{project}/{freqmode}/comments/".format(
+            host="http://localhost",
             version="v4",
             project=PROJECT_NAME,
             freqmode=urlinfo.freq_mode,
         )
-        r = requests.get(url)
+        r = test_client.get(
+            url,
+            follow_redirects=True,
+        )
         assert r.status_code == http.client.OK
-        comments = r.json()["Info"]["Comments"]
+        comments = r.json["Info"]["Comments"]
         assert len(comments) == 6
 
-    def test_get_comments_v5(self, fake_data, odinapi_service):
+    def test_get_comments_v5(self, fake_data, test_client):
         urlinfo, _ = fake_data
-        url = "{host}/rest_api/{version}/level2/{project}/{freqmode}/comments".format(
-            host=odinapi_service,
+        url = "{host}/rest_api/{version}/level2/{project}/{freqmode}/comments/".format(
+            host="http://localhost",
             version="v5",
             project=PROJECT_NAME,
             freqmode=urlinfo.freq_mode,
         )
-        r = requests.get(make_dev_url(url))
+        r = test_client.get(
+            make_dev_url(url),
+            follow_redirects=True,
+        )
         assert r.status_code == http.client.OK
-        assert r.json()["Type"] == "level2_scan_comment"
-        comments = r.json()["Data"]
+        assert r.json["Type"] == "level2_scan_comment"
+        comments = r.json["Data"]
         assert len(comments) == 6
 
-    def test_get_comments_v5_respects_limit(self, fake_data, odinapi_service):
+    def test_get_comments_v5_respects_limit(self, fake_data, test_client):
         urlinfo, _ = fake_data
-        url = "{host}/rest_api/{version}/level2/{project}/{freqmode}/comments?limit=1".format(  # noqa
-            host=odinapi_service,
+        url = "{host}/rest_api/{version}/level2/{project}/{freqmode}/comments/?limit=1".format(  # noqa
+            host="http://localhost",
             version="v5",
             project=PROJECT_NAME,
             freqmode=urlinfo.freq_mode,
         )
-        r = requests.get(make_dev_url(url))
+        r = test_client.get(
+            make_dev_url(url),
+            follow_redirects=True,
+        )
         assert r.status_code == http.client.OK
-        comments = r.json()["Data"]
+        comments = r.json["Data"]
         assert len(comments) == 1
         assert comments[0]["Comment"] == "Error: This scan failed"
 
-    def test_get_comments_v5_respects_offset(self, fake_data, odinapi_service):
+    def test_get_comments_v5_respects_offset(self, fake_data, test_client):
         urlinfo, _ = fake_data
-        url = "{host}/rest_api/{version}/level2/{project}/{freqmode}/comments?offset=5".format(  # noqa
-            host=odinapi_service,
+        url = "{host}/rest_api/{version}/level2/{project}/{freqmode}/comments/?offset=5".format(  # noqa
```

Comment:
This is the correct URL, with trailing slash? If so, we should make sure the URLs returned in responses (Scan Log etc.) are correct, so that we don't get unnecessary redirects and similar in processing.

Reply:
We have sort of a  problem with some endpoints ending with slash and some the without. requests follow them automatically test_client don't. I'll have an extra look.